src/main/java/io/r2dbc/h2/codecs/BigDecimalCodec.java

```diff
@@ -30,7 +30,7 @@
 
     @Override
     boolean doCanDecode(int dataType) {
-        return dataType == Value.DECIMAL;
+        return dataType == Value.NUMERIC;
```

Comment:
`|| dataType == Value.DECFLOAT` may be needed too. But `DECFLOAT` has three special values (`NaN`, `Inf`, and `-Inf`) not compatible with `BigDecimal`, so situation is complicated.

Reply:
I opened https://github.com/r2dbc/r2dbc-h2/issues/209 to track this.